.github/workflows/build-and-test-pkg.yaml

```diff
@@ -35,316 +29,59 @@ jobs:
             tag=${{ github.ref_name }}
           fi
           echo "tag=$tag" >> ${GITHUB_OUTPUT}
-  
+
   macos-aarch64-pkg-build:
     needs: get-tag-name
-    runs-on: [self-hosted, macos, arm64, 14, release]
-    timeout-minutes: 60
-    steps:
-      - uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
-        with:
-          ref: ${{ needs.get-tag-name.outputs.tag }}
-          fetch-depth: 0
-          persist-credentials: false
-          submodules: true
-      - uses: actions/setup-go@cdcb36043654635271a94b9a6d1392de5bb323a7 # v5.0.1
-        with:
-          go-version-file: go.mod
-          cache: true
-      - name: Make macos aarch64 build
-        run: |
-          brew install lz4 automake autoconf libtool yq
-          git status
-          git clean -f -d
-          make clean
-          make download-licenses 
-          make FINCH_OS_IMAGE_LOCATION_ROOT=/Applications/Finch
-        shell: zsh {0}
-
-      - name: configure aws credentials
-        uses: aws-actions/configure-aws-credentials@e3dd6a429d7300a6a4c196c26e071d42e0343502 # v4.0.2
-        with:
-          role-to-assume: ${{ secrets.ROLE }}
-          role-session-name: dependency-upload-session
-          aws-region: ${{ secrets.REGION }}
-
-      - name: generate pkg
-        run: |
-          ./installer-builder/tools/release-installer.sh aarch64 ${{ needs.get-tag-name.outputs.tag }} ${{ secrets.INSTALLER_PRIVATE_BUCKET_NAME }} ${{ secrets.EXECUTABLE_BUCKET }} ${{ secrets.PKG_BUCKET }} ${{ secrets.NOTARIZATION_ACCOUNT }} ${{ secrets.NOTARIZATION_CREDENTIAL }}
-        shell: zsh {0}
+    uses: ./.github/workflows/build-pkg.yaml
+    secrets: inherit
+    with:
+      os: macos
+      arch: arm64
+      output_arch: aarch64
+      version: 14
```

Comment:
Curious why we are not building for the other versions (11,13) but we are testing it later.

Reply:
Finch should run on multiple versions of macOS, but you only have to build the executable once per architecture in order to support that, thanks to the changes to the Makefile in https://github.com/runfinch/finch/pull/947.